ls1intum/Artemis
# `Programming exercises`: Highlight code issues in short result information

### Checklist
#### General


#### Client

### Motivation and Context
PR #6460 changed that the result string appear by default in the short format. It might be the case, though, that a student has passed most/all test cases, but still has code style/static code analysis issues that should be resolved. The short format gives does not indicate that in any way without clicking on it to inspect the detailed feedback.

### Description
To indicate that code issues exist while keeping the result string short, a warning icon is added to the short format of the result string.
In the long format no icon is used since the string already indicates the concrete number of issues.

The icon shows a tooltip on hover that mentions the code analysis warnings. With that combined with the identical colour the necessary mental connection between this part of the result string and the relevant section (‘Warnings’) in the detailed feedback should be clear for students.
 
### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Programming Exercise with static code analysis

1. Log in to Artemis
2. Edit the solution participation code to introduce a static code analysis issue.
3. All tests pass, the result shows 100%, but the result string contains the warning icon as shown in the screenshots below. On hover over the icon, a translated tooltip is shown.

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage




| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| `result.component.ts` | 100% for the relevant cases in the HTML template | ✅              |


### Screenshots



Warning icon in case of SCA issues regardless of score:
![grafik](https://github.com/ls1intum/Artemis/assets/64250573/f4f82b70-f310-40ad-a1de-044a61d1a80c)
![grafik](https://github.com/ls1intum/Artemis/assets/64250573/cb327f4b-6aee-4342-87a4-666fe28d2cda)
![grafik](https://github.com/ls1intum/Artemis/assets/64250573/9ee5b1bd-d877-4cb4-9975-5402d2a323e2)
Long result variant has no icon since the text already provides the information:
![Screenshot 2023-05-17 at 09-05-18 Programming Exercises](https://github.com/ls1intum/Artemis/assets/64250573/4e8d906c-4265-46b4-87ca-e446fdf1efd3)

<details>
<summary>Iteration 1: Colour change</summary>

With issues:
![image](https://github.com/ls1intum/Artemis/assets/64250573/d722ab51-8308-442f-bccc-1f5fdf23c72a)
![Screenshot 2023-05-15 at 17-04-58 Code Editor](https://github.com/ls1intum/Artemis/assets/64250573/3ac94246-24cb-4dce-a62b-0e44f0feee04)

Without issues:
![image](https://github.com/ls1intum/Artemis/assets/64250573/6932c9be-b0ec-4c24-8a11-ce0487521353)


0% still red:
![image](https://github.com/ls1intum/Artemis/assets/64250573/f1291ed2-9261-488e-8524-afbc3d178682)
</details>
